specs/PushNotifications/PushNotifications-spec.md

```diff
@@ -0,0 +1,438 @@
+            ComponentUI = 1007,
+            ProtocolForResults = 1009,
+            ToastNotification = 1010,
+            Print3DWorkflow = 1011,
+            DialReceiver = 1012,
+            DevicePairing = 1013,
+            UserDataAccountsProvider = 1014,
+            FilePickerExperience = 1015,
+            LockScreenComponent = 1016,
+            ContactPanel = 1017,
+            PrintWorkflowForegroundTask = 1018,
+            GameUIProvider = 1019,
+            StartupTask = 1020,
+            CommandLineLaunch = 1021,
+            BarcodeScannerProvider = 1022,
+            Push = 1023,
+        };
+
+        // Simply a wrapper around IActivatedEventArgs to support new Kind extensions like Push which aren't a part of Foundations today
+        runtimeclass ActivatedEventArgsExtension
+        {
+            ActivatedEventArgsExtension(Windows.ApplicationModel.Activation.IActivatedEventArgs activatedEventArgs, ActivationKindExtension kind2);
+            ActivationKindExtension Kind2 { get; };
+            Windows.ApplicationModel.Activation.IActivatedEventArgs ActivatedArgs{ get; };
+        }
+
+        // This is an overload of the existing GetActivatedEventArgs in Reunion which supports new Activation Kinds
+        static runtimeclass AppLifecycle2
+        {
+            // Called during process launch in Main to determine whether the launch is in response to Protocol/Foreground/Push etc.
+            static ActivatedEventArgsExtension GetActivatedEventArgs();
+        }
+
+        // Event args in WinMain activation payload. It wraps around the BackgroundInstance construct and manages the task lifetime.
+        runtimeclass PushNotificationActivatedEventArgs : Windows.ApplicationModel.Activation.IActivatedEventArgs
+        {
+            // Initialize using the raw byte payload
+            PushNotificationActivatedEventArgs(Windows.ApplicationModel.Background.IBackgroundTaskInstance backgroundTask);
+
+            // The Push payload
+            byte[] Payload { get; };
+
+            // Takes a deferral to run under specific modes like low power mode
+            void GetDeferral();
+
+            // Closes the Deferral taken on a Push Trigger. If this is not called by the developer, the deferral will automatically be invoked on destruction.
+            void CompleteDeferral();
+
+            // Subscribe to Cancelled event handler to be signalled when resource policies are no longer true like 30s wallclock timer
+            event Windows.ApplicationModel.Background.BackgroundTaskCanceledEventHandler Canceled;
+
+            // The number of times Resource management suspended the Task
+            UInt32 SuspendedCount { get; };
+        }
+
+        // An abstraction over the inproc activation Registration flow
+        runtimeclass InProcActivatorDetails
+        {
+            // Initialize using a manifest defined activatorId to Activate the InProc COM server
+            InProcActivatorDetails(String taskClsid);
```

Comment:
Nit: Can we use a GUID type here so there's an expected form? Right now it's not clear if I can pass in something with or without brackets. (e.g. `Windows.ApplicationModel.Background.SetTaskEntryPointClsid([in] GUID TaskEntryPoint)`)

Reply:
Good point. Will change.